hivemind/utils/asyncio.py

```diff
@@ -42,3 +67,26 @@ async def await_cancelled(awaitable: Awaitable) -> bool:
         return True
     except BaseException:
         return False
+
+
+async def async_map(func: Callable[..., T], *iterables: AsyncIterable, max_prefetch: Optional[int] = None,
+                    executor: Optional[ThreadPoolExecutor] = None) -> AsyncIterator[T]:
+    """ iterate from an async iterable in a background thread, yield results to async iterable """
```

Comment:
I think that it's not map but starmap, since you pass multiple iterables to it :) 
Two options — either rename, or create a map-only version and use it as starmap only where needed (passing azip as an argument)